File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        unordered_set<int> seen;
        for (int x : nums) {
            if (x % k == 0) {
                seen.insert(x / k);
            }
        }
        int i = 1;
        while (seen.count(i)) {
            ++i;
        }
        return i * k;
    }
};

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.ts
function missingMultiple(nums: number[], k: number): number {
    const seen = new Set<number>();
    for (const x of nums) {
        if (x % k === 0) {
            seen.add(x / k);
        }
    }
    let i = 1;
    while (seen.has(i)) {
        i++;
    }
    return i * k;
}